### PR TITLE
f-content-cards@v2.2.0-beta.2 - Adds groupCards option / prop to content cards to display by group title

### PR DESCRIPTION
**Pull request by matthewhardern**
Adds new callback for groups, function for handling grouped card impressions, plus new function for handling content cards. 
Adds template changes to support looping over groups.

Minor style changes to make cards line up both in height and across with margin in the design. ( can be removed but needed it to see visually what I am doing)  

- [x] README and/or UI Documentation has been [created|updated]
- [x] Unit tests have been [created|updated]

As a side note I am not running browser tests as part of this PR as a this will be handled in my next PR for this component with PIE changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+v2.9.0
+------------------------------
+*October 26, 2020*
+
+### Added
+- Added an entry to gitignore to ignore some local development tools for mocking http requests
```
Comment (minrwhite) [inline CHANGELOG.md:12]: Add an extra newline between version entries, please

File: packages/f-content-cards/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,15 @@
...
+------------------------------
+*October 26, 2020*
+
+### Added
+- Integrated card grouping via new Braze dispatcher method, added new prop groupCards and added new
+watch for cardsGrouped enabling a new emitter containing the number of groups and logCardImpressions
+when grouped as well. Please be aware while groupedCards is true limiting cannot be used, this is until
+we have decided how to apply limiting to groups.
 
 Latest (to be added to next release)
```
Comment (minrwhite) [inline packages/f-content-cards/CHANGELOG.md]: Concatenate the entries below into the one above.

File: packages/f-content-cards/src/components/ContentCards.vue [hunk 3]
```diff
@@ -3,16 +3,40 @@
...
+                <div
+                    v-for="(contentCard, cardIndex) in subCards"
+                    :key="`${groupIndex}_${cardIndex}_${contentCard.id}`"
+                    :class="[$style['c-contentCards--group']]"
+                >
+                    <component
+                        :is="handleCustomCardType(contentCard.type)"
+                        :card="contentCard"
+                        :tenant="tenant"
+                        :data-test-id="testIdForItemWithIndex(cardIndex)"
```
Comment (minrwhite) [inline packages/f-content-cards/src/components/ContentCards.vue]: We probably should do something different with grouped cards, as the same index is certain to reoccur between different groups of cards, leading to cards with identical test ids. Maybe modifying the `testIdForItemWithIndex()` function to accept an optional `groupIndex` parameter also, could solve this.

File: packages/f-content-cards/src/components/ContentCards.vue [hunk 4]
```diff
@@ -3,16 +3,40 @@
...
-                :title="title"
-                :tenant="tenant"
-                :data-test-id="testIdForItemWithIndex(cardIndex)"
-            />
+        <template v-if="groupCards">
+            <template v-for="({ title, cards: subCards, id: groupId }, groupIndex) in cardsGrouped">
+                <h3
+                    v-if="groupId"
+                    :key="groupIndex"
+                    :class="[$style['c-contentCards--group-title']]"
```
Comment (minrwhite) [inline packages/f-content-cards/src/components/ContentCards.vue]: Can we add a non-`$style` wrapped `'c-contentCards--group-title'` class here, to allow for externally introduced styling, please?

File: packages/f-content-cards/src/components/cardTemplates/CardContainer.vue [hunk 5]
```diff
@@ -253,13 +258,14 @@ export default {
...
         display: flex;
+        flex-grow: 1;
         flex-direction: column;
         align-items: center;
         min-height: 164px; // min-height set to the height of an card with a one-line title
         background-color: $white;
         padding: spacing(x3) spacing(x2);
         box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.1);
-        height: 100%;
+        //height: 100%;
```
Comment (minrwhite) [inline packages/f-content-cards/src/components/cardTemplates/CardContainer.vue]: Remove rather than comment please, if this isn't needed.

Comment (dandel10n) [inline packages/f-content-cards/src/components/cardTemplates/CardContainer.vue]: Should this be removed if not needed anymore?

File: packages/f-content-cards/stories/mockData/cards.js [hunk 6]
```diff
@@ -101,7 +101,7 @@ const fieldTypeToFaker = {
 export const labelledMultiSelectAllowedValues = Object.fromEntries(Object.entries(cardTypes)
     .map(([key, { label }]) => [label, key]));
 
-function timeRange10HoursAroundNow () {
+export function timeRange10HoursAroundNow () {
```
Comment (minrwhite) [inline packages/f-content-cards/stories/mockData/cards.js]: Do we need this exported?

File: packages/f-content-cards/src/components/ContentCards.vue (hunk 3 above)
Comment (matthewhardern) [reply packages/f-content-cards/src/components/ContentCards.vue]: Thanks missed this one, will add optional parameter and handle it as suggested.

File: packages/f-content-cards/src/components/ContentCards.vue (hunk 4 above)
Comment (matthewhardern) [reply packages/f-content-cards/src/components/ContentCards.vue]: will do

File: packages/f-content-cards/src/components/cardTemplates/CardContainer.vue (hunk 5 above)
Comment (matthewhardern) [reply packages/f-content-cards/src/components/cardTemplates/CardContainer.vue]: Yep my mistake, removing

File: packages/f-content-cards/stories/mockData/cards.js (hunk 6 above)
Comment (matthewhardern) [reply packages/f-content-cards/stories/mockData/cards.js]: No we don't it was exported as part of some local development work with mocking the API's. I shall remove it and put it back to previous state.

File: packages/f-content-cards/src/components/cardTemplates/CardContainer.vue (hunk 5 above)
Comment (matthewhardern) [reply packages/f-content-cards/src/components/cardTemplates/CardContainer.vue]: fixed

File: packages/f-content-cards/stories/mockData/cards.js (hunk 6 above)
Comment (matthewhardern) [reply packages/f-content-cards/stories/mockData/cards.js]: removed

File: packages/f-content-cards/src/components/ContentCards.vue (hunk 4 above)
Comment (matthewhardern) [reply packages/f-content-cards/src/components/ContentCards.vue]: added to allow external styles

File: packages/f-content-cards/src/components/ContentCards.vue (hunk 3 above)
Comment (matthewhardern) [reply packages/f-content-cards/src/components/ContentCards.vue]: added a optional parameter and additional test

File: packages/f-content-cards/CHANGELOG.md (hunk 2 above)
Comment (matthewhardern) [reply packages/f-content-cards/CHANGELOG.md]: they are now on one line

File: CHANGELOG.md (hunk 1 above)
Comment (matthewhardern) [reply CHANGELOG.md:12]: added the extra new line

File: packages/f-content-cards/src/components/cardTemplates/CardContainer.vue (hunk 5 above)
Comment (matthewhardern) [reply packages/f-content-cards/src/components/cardTemplates/CardContainer.vue]: thanks have removed

File: packages/f-content-cards/CHANGELOG.md (hunk 2 above)
Comment (minrwhite) [reply packages/f-content-cards/CHANGELOG.md]: Thanks, but that's not quite what I meant - everything below the `Latest (to be added to next release)` should be added into the `v2.2.0-beta.2` changelog; `Added` entries added on their own lines alongside those that exist, `Changed` entry added on its own.

File: packages/f-content-cards/src/components/tests/ContentCards.test.js [hunk 7]
```diff
@@ -822,5 +822,31 @@ describe('ContentCards', () => {
             // Assert
             expect(metadataDispatcher.logCardImpressions).toHaveBeenCalledTimes(1);
         });
+
+        it('should generate test id attributes on child content cards components within a group', async () => {
```
Comment (minrwhite) [inline packages/f-content-cards/src/components/tests/ContentCards.test.js:826]: 💯

File: packages/f-content-cards/CHANGELOG.md (hunk 2 above)
Comment (matthewhardern) [reply packages/f-content-cards/CHANGELOG.md]: Sorry for the confusion, get what you mean now sorry :)

**Review by minrwhite — DISMISSED** [review]

File: packages/f-content-cards/CHANGELOG.md (hunk 2 above)
Comment (matthewhardern) [reply packages/f-content-cards/CHANGELOG.md]: changed

**Review by dandel10n — APPROVED** [review]

**Review by dandel10n — CHANGES_REQUESTED** [review]
Noticed that you haven't increased the version in package.json in the project root to 2.9.0 as noted in changelog. Actually, because it is only gitignore change I am not sure if the version bump is needed at all. Maybe it is better to leave a note in changelog "Latest (to be added to next release)" instead of "v2.9.0" on line 6.

**Comment by matthewhardern** [comment]
@dandel10n agreed, change made to the changelog. 👍

**Review by minrwhite — APPROVED** [review]

**Review by dandel10n — APPROVED** [review]

**Pull request merged**